Neo4j.Driver/Neo4j.Driver/Internal/Temporal/TimeZoneMapping.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) 2002-2018 "Neo Technology,"
+// Network Engine for Objects in Lund AB [http://neotechnology.com]
+// 
+// This file is part of Neo4j.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Xml.Linq;
+
+namespace Neo4j.Driver.Internal.Temporal
+{
+    internal static class TimeZoneMapping
+    {
+        private static readonly IDictionary<string, string> IANAToWindows;
+        private static readonly IDictionary<string, string> WindowsToIANA;
+        private static readonly IDictionary<string, TimeZoneInfo> SystemToTZInfo;
+
+        static TimeZoneMapping()
+        {
+            try
+            {
+                var typeInfo = typeof(TimeZoneMapping).GetTypeInfo();
+                var resource =
+                    typeInfo.Assembly.GetManifestResourceStream("Neo4j.Driver.Internal.Temporal.windowsZones.xml");
+
+                Load(resource, out var ianaToWindows, out var windowsToIana);
+
+                IANAToWindows = new ReadOnlyDictionary<string, string>(ianaToWindows);
+                WindowsToIANA = new ReadOnlyDictionary<string, string>(windowsToIana);
+            }
+            catch (Exception)
+            {
+                // We don't want this to break the program
+                IANAToWindows = new ReadOnlyDictionary<string, string>(new Dictionary<string, string>());
+                WindowsToIANA = IANAToWindows;
+            }
+            finally
+            {
+                SystemToTZInfo =
+                    new ReadOnlyDictionary<string, TimeZoneInfo>(TimeZoneInfo.GetSystemTimeZones()
+                        .ToDictionary(tz => tz.Id, tz => tz));
+            }
+        }
+
+        public static TimeZoneInfo Get(string zoneId)
```

Comment:
There are paths:
1) `if (SystemToTZInfo.TryGetValue(zoneId, out var tzInfo))`
2) `if (IANAToWindows.TryGetValue(zoneId, out var windowsId))`
3) `if (WindowsToIANA.TryGetValue($"{zoneId}_{GetCurrentTerritory()}", out var ianaId) ||`
4) `return TimeZoneInfo.FindSystemTimeZoneById(zoneId);`
5) error at step 4)

Could you make sure they are all covered by some unit tests to prevent anyone ever break this logic by mistake here?

The current unit tests looks only covers one path if I understand correctly?

Reply:
@zhenlineo I'll add tests for other paths as another PR.